### PR TITLE
Add ARPACK-NG information

### PR DESCRIPTION
**Pull request by masterleinad**
After having to install `ARPACK` manually again, I noticed how much easier this is using `ARPACK-NG` instead. In particular, `ARPACK-NG` has a CMake build system.

### CODE REVIEW

**Comment by drwells** [comment]
This looks good: I was able to follow the directions and now I have compiled and installed ARPACK :)

Is there `make test` or `make check` function implemented in the CMake version of their build system?

**Comment by drwells** [comment]
@davydden You have my respect for taking the time to write patches for this venerable eigenvalue library :)

**Comment by davydden** [comment]
@drwells i did not write that many, but this one https://github.com/opencollab/arpack-ng/pull/21 was a nightmare to come up with starting from the segmentation fault in the ScaLAPACK library :smile:

**Comment by masterleinad** [comment]
@drwells Yes, there is both `make check` and `make test`. I'll add a remark.

File: doc/external-libs/arpack.html [hunk 1]
```diff
@@ -24,7 +24,9 @@
 
     <p>
       Below is a short summary of instructions on how to compile and install
-      <acronym>ARPACK</acronym> by hand (for the case you wish to do so).
+      <acronym>ARPACK</acronym> by hand (for the case you wish to do so)
+      or using <a href="https://github.com/opencollab/arpack-ng">ARPACK-NG</a>
+      that equips <acronym>ARPACK</acronym> with a CMake build system.
```
Comment (bangerth) [inline doc/external-libs/arpack.html]: if arpack-ng also has bug fixes, maybe that is worth mentioning as well.

File: doc/external-libs/arpack.html [hunk 2]
```diff
@@ -54,26 +55,63 @@
...
     <p>
-    Note: For compilation of <acronym>ARPACK</acronym> we emphasise
-    adding the compiler flag <code>-fPIC</code>. This is a definite
-    requirement if we are compiling <acronym>deal.II</acronym> with
-    shared libraries (which is the default). If we had preferred to be
-    compiling <acronym>deal.II</acronym> without shared libraries,
-    that's ok too; in that case we would do exactly the same thing
-    as described above, but this time omitting
-    the <code>-fPIC</code> flag from the scheme.
+      Note: For compilation of <acronym>ARPACK</acronym> we emphasise
```
Comment (bangerth) [inline doc/external-libs/arpack.html]: -> emphasize (while you're there already)

File: doc/external-libs/arpack.html (hunk 1 above)
Comment (masterleinad) [inline doc/external-libs/arpack.html]: Now, I am just linking to their `CHANGES` page.

**Pull request merged**